pkg/project/registry/projectrequest/delegated/delegated_test.go

```diff
@@ -0,0 +1,88 @@
+package delegated
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	rbaclisters "k8s.io/kubernetes/pkg/client/listers/rbac/internalversion"
+
+	"github.com/go-openapi/errors"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+)
+
+func TestDelegatedWait(t *testing.T) {
+	cache := &testRoleBindingLister{}
+	storage := &REST{roleBindings: cache}
+
+	cache.namespacelister = &testRoleBindingNamespaceLister{}
+	cache.namespacelister.bindings = map[string]*rbac.RoleBinding{}
+	cache.namespacelister.bindings["anything"] = nil
+
+	callReturnedCh := testWait(storage)
+
+	select {
+	case <-callReturnedCh:
+		t.Errorf("too fast, should have blocked")
+	case <-time.After(1 * time.Second):
+	}
+
+	func() {
+		cache.namespacelister.lock.Lock()
+		defer cache.namespacelister.lock.Unlock()
+		cache.namespacelister.bindings[bootstrappolicy.AdminRoleName] = &rbac.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{Name: bootstrappolicy.AdminRoleName},
```

Comment:
Is it possible to extract this to a function in the `testRoleBindingLister` struct?